### PR TITLE
fix: using our own pod template hashing

### PR DESCRIPTION
**Pull request by perenesenko**
Fix for https://github.com/argoproj/argo-rollouts/issues/1356

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: utils/hash/hash.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+// ComputePodTemplateHash returns a hash value calculated from the Rollout's pod template.
```
Comment (huikang) [inline utils/hash/hash.go:13]: nit: "from the Rollout's pod template." -> ""from the pod template as the podtemplate could be from an experiment

**Review by huikang — COMMENTED** [review]
Looks great, except for some failed unit test.

File: utils/hash/hash.go (hunk 1 above)
Comment (perenesenko) [reply utils/hash/hash.go]: fixed

**Review by huikang — APPROVED** [review]
LGTM

**Pull request merged**